libs/api/user/src/lib/adapters/prisma-user.adapter.ts

```diff
@@ -0,0 +1,64 @@
+import { Injectable } from '@nestjs/common';
+import { OrmPrismaService } from '@cypher/api/shared/database/orm';
+import { UserRepository } from '../user.repository';
+import { UserObjectType } from '../user.model';
+import { UserUniqueFields } from '../types';
+
+@Injectable()
+export class PrismaUserAdapter implements UserRepository {
+  constructor(private readonly prisma: OrmPrismaService) {}
+
+  async getUsers(
+    key: UserUniqueFields,
+    values: string[]
+  ): Promise<UserObjectType[]> {
+    return await this.prisma.user.findMany({
+      where: {
+        [key]: {
+          in: values,
+        },
+      },
+    });
+  }
+
+  async follow(followed: string, following: string): Promise<UserObjectType> {
```

Comment:
Are you sure about the return type ? Here you return a `Prisma.User` but yoru return type is `UserObjectType`.

Reply:
Hey Thomas ! The point is that we don't want to return a Prisma.User but a UserObjectType since our adapter implements the repository. The Prisma.User looks like a UserObjectType so TS is ok with that.